test.py

```diff
@@ -116,9 +116,13 @@ def test_user_becomes_present(edap):
 
 
 def test_it_division_becomes_present(edap):
+    description = b"It division"
     assert not edap.object_exists_at(f"cn=it,{edap.DIVISIONS_GROUP}", "posixGroup")
-    edap.create_division("it")
+    edap.create_division("it", display_name=description)
     assert edap.object_exists_at(f"cn=it,{edap.DIVISIONS_GROUP}", "posixGroup")
+    res = edap.get_division("it")
+    assert len(res) == 1
+    assert res[0][1]['description'][0] == description
```

Comment:
The `get_division` should be refactored - the name suggests that one would expect some kind of division data structure, and perhaps even an exception when there is nothing.
It would be great to simplify the test line to `description in res[0][1]['description']` - use the `in` operator to get rid of the trailing `[0]`, and expand the `get_division` function to pass just `attributes` from the search result, which is supposed to be exactly one division.

Reply:
So should I implement some class for Division class in this PR or leave it as it is for now?
Yes, agree, will probably refactor `get_user`, `get_group` the same way too.

Reply:
By the way, ldap always returns a list of values for every attribute, but does values for `description`, `name` and some other attributes can have more than one value there ? Or may be we could try to unpack such elements? Because even on front-end in TEAP there is always a case like `item['description'][0]`

Reply:
For now, added raising Exception if no object found and suggested change in tests.

Reply:
I think that a class for division may be too much at the moment, but I like the `get_single_object` method. However, the LDAP search result returns a (dn, attributes) tuple, and I think that we may drop the `dn`, and return only those attributes.

Reply:
@matejak you mean drop it from everywhere or only in get_single_object method ? 
But won't be a bit complicated working with such objects after, because we will need to construct dn manually ?

Reply:
@danil-topc You are right that it may be a bit short-sighted to drop dn completely, but the current state is very inconvenient.
You can go for a compromise - add a `fqdn` attribute, and get it from the returned `DN`. That way, it won't be lost, but dealing with search results will be much easier.
The point is that the Python LDAP module has it's interface, and we shouldn't feel like we have to use it. We can come up to our way how to represent objects, so it is easy to work with them. Representing an object as a dict has the advantage that it is easy to work with, and it is very close to the Python LDAP representation as well.
And feel free to get rid of the tuple everywhere - the `get_single_object` just "knows" that there is at most one thing to return.

Reply:
@matejak looks like a good compromise, check last commit please, added transformation from list of tuples to list of dicts, adding dn as fqdn to attributes. If it's ok, will also add fixes to TEAP project